pages/studentPortal/index.tsx

```diff
@@ -65,6 +72,37 @@ export default function StudentPortalPage() {
     });
   }, [user]);
 
+  const [updateLastSeenModal] = useMutation(UPSERT_LAST_SEEN_MODAL);
+  const [showModal, setShowModal] = useState(false);
+
+  useQuery<FetchModalData>(FETCH_LAST_SEEN_MODAL, {
+    variables: {
+      _id: user.uid,
+    },
+
+    onCompleted: (data) => {
+      if (data.users[0].freemium_user) {
+        const lastSeenValue = data.users[0].freemium_user?.lastSeenModal;
+        const lastSeenDifference = differenceInSeconds(
+          new Date(),
+          new Date(lastSeenValue)
+        );
+
+        if (lastSeenDifference > 60) {
```

Comment:
will this show a user the freemium modal every minute?

Reply:
I adjusted to 24 hours.